crates/wasi-common/yanix/src/sys/bsd/file.rs

```diff
@@ -8,9 +8,13 @@ pub unsafe fn isatty(fd: RawFd) -> Result<bool> {
         Ok(true)
     } else {
         // ... otherwise 0 is returned, and errno is set to indicate the error.
-        let errno = Errno::last();
-        if errno == Errno::ENOTTY {
-            Ok(false)
+        let errno = io::Error::last_os_error();
+        if let Some(raw_errno) = errno.raw_os_error() {
```

Comment:
`raw_os_error` will [always](https://doc.rust-lang.org/std/io/struct.Error.html#method.raw_os_error) return `Some` when the error comes from `last_os_error`, so you can just `unwrap()` it here, and same for the other places in the code where this comes up.

Reply:
Makes sense, I'll make the adjustments. This should clean up the calls since this way we'll effectively get rid of one level of nesting.

Reply:
Done in 686fd9d. Hopefully I got all of them!